pkg/ccl/backupccl/restore_planning.go

```diff
@@ -1462,6 +1471,12 @@ func checkBackupManifestVersionCompatability(
 	minimumRestoreableVersion := p.ExecCfg().Settings.Version.BinaryMinSupportedVersion()
 	for i := range mainBackupManifests {
 		v := mainBackupManifests[i].ClusterVersion
+
+		// Allow restores from the serverless 23.1 preview cluster onto 23.1
+		if unsafeRestoreIncompatibleVersion && serverlessMRPreviewVersion.Equal(v) {
```

Comment:
I'm somewhat inclined to just remove `&& serverlessMRPreviewVersion` and just make `unsafeRestoreIncompatibleVersion` trump both limit checks instead of just the one. I guess at that point we'd just move it to be an early return before the loop.